engine/runtime/src/main/java/org/enso/interpreter/runtime/number/EnsoBigInteger.java

```diff
@@ -0,0 +1,22 @@
+package org.enso.interpreter.runtime.number;
+
+import java.math.BigInteger;
+
+/** Internal wrapper for a {@link BigInteger}. */
+public class EnsoBigInteger {
+  private final BigInteger value;
+
+  /**
+   * Wraps a {@link BigInteger}.
+   *
+   * @param value the value to wrap.
+   */
+  public EnsoBigInteger(BigInteger value) {
```

Comment:
When passing a big integer over interop what happens? Do we do a check and wrap it into this number system?

Reply:
Nothing happens. There's no Interop protocol for big numbers, so a big int is just "an object" that will never be casted back to our number system. I will be doing some work on casting other number types tho (like `int`s from java will become longs etc.) in a later stage of this task.